R/dt_boxhead.R

```diff
@@ -70,14 +70,61 @@ dt_boxhead_edit <- function(data, var, ...) {
   dt_boxhead %>% dt_boxhead_set(data = data)
 }
 
+dt_boxhead_add_var <- function(data,
+                               var,
+                               type,
+                               column_label = list(var),
+                               column_align = "left",
+                               column_width = list(NULL),
+                               hidden_px = list(NULL),
+                               add_where = "top") {
+
+  dt_boxhead <- data %>% dt_boxhead_get()
+
+  dt_boxhead_row <-
+    dplyr::tibble(
+      var = var,
+      type = type,
+      column_label = column_label,
+      column_align = column_align,
+      column_width = column_width,
+      hidden_px = hidden_px
+    )
+
+  if (add_where == "top") {
+    dt_boxhead <- dplyr::bind_rows(dt_boxhead_row, dt_boxhead)
+  } else if (add_where == "bottom") {
+    dt_boxhead <- dplyr::bind_rows(dt_boxhead, dt_boxhead_row)
+  } else {
+    stop("The `add_where` value must be either `top` or `bottom`.")
+  }
+
+  dt_boxhead %>% dt_boxhead_set(data = data)
+}
+
 dt_boxhead_set_hidden <- function(data, vars) {
 
   dt_boxhead <- dt_boxhead_get(data = data)
 
```

Comment:
I won't make you change it now, but it would be nice for functions like this to throw an error unless `all(vars %in% dt_boxhead$var)`